sql/parser/sql.y

```diff
@@ -1402,11 +1402,11 @@ rename_stmt:
   }
 | ALTER INDEX qualified_name RENAME TO name
   {
-    $$ = nil
+    $$ = &RenameIndex{Name: $3, NewName: Name($6), IfExists: false}
   }
```

Comment:
Confusingly, I believe the grammar support already exists: `ALTER TABLE database.table.index RENAME TO new_index_name`. So the `RenameTable` struct could really be thought of as `RenameTableOrIndex` (though that name is overly verbose).